src/main/scala/cromwell/engine/backend/jes/Run.scala

```diff
@@ -25,7 +25,10 @@ object Run  {
 
   def apply(pipeline: Pipeline): Run = {
     val rpr = new RunPipelineRequest().setPipelineId(pipeline.id).setProjectId(pipeline.projectId).setServiceAccount(JesServiceAccount)
-    val tag = s"JES Run [UUID(${pipeline.workflow.shortId}):${pipeline.key.scope.name}]"
+    def tag: String = {
+      val shard = pipeline.key.index.map(x => s":$x").getOrElse("")
+      s"JES Run [UUID(${pipeline.workflow.shortId}):${pipeline.key.scope.name}$shard]"
+    }
```

Comment:
I don't know if I agree with the `toString` on IndexEnhancedIndex... one reason is that `toString` should probably be more descriptive than just `:4` or `` (maybe `shard 4` and `no shard` are better toString values?)  but even if we call it `tagPartial` or `tag` or something, then it's only ever being called in this one place (well, in `ExecutionStoreKey.tag`).

Maybe a second reviewer opinion?